### PR TITLE
MB-58033: Support for custom datetime layouts

### PR DESCRIPTION
**Pull request by CascadingRadium**
## Jira

[MB-58033](https://issues.couchbase.com/browse/MB-58033)

## Description
- Add 'percentstyle' date time layout format that mimics 'strftime' used for date time layout specification in C and Python.
- Add 'isostyle' date time layout format that attempts to mimic 'DateTimeFormatter' which is used to specify date time layouts in Java.
- Added unit tests

### CODE REVIEW

File: analysis/datetime/percent/percent.go
Comment (Thejas-bhat) [inline analysis/datetime/percent/percent.go:1]: needs to 2023

**Review by Thejas-bhat — CHANGES_REQUESTED** [review]

File: analysis/datetime/percent/percent.go
Comment (abhinavdangeti) [inline analysis/datetime/percent/percent.go:26]: Wondering if this should have the "go" suffix. `flexiblego` and `sanitizedgo` are for supported golang formats. This really isn't.

**Review by abhinavdangeti — COMMENTED** [review]
Can you quickly set up a separate document highlighting all the support this will be adding, which we can later share with the docs team.

**Review by abhinavdangeti — DISMISSED** [review]
Please wait for one more review here.

**Review by Thejas-bhat — APPROVED** [review]

File: analysis/datetime/javatime/javatime.go
Comment (abhinavdangeti) [inline analysis/datetime/javatime/javatime.go:26]: We're not going to call this "javastyle". Drop all references to "java" from function names and file names.

You can indicate in comments that you got it from oracle but choose a better name for this.

**Review by abhinavdangeti — CHANGES_REQUESTED** [review]

**Comment by abhinavdangeti** [comment]
Also, not sure how you rebase this, but it seems you've cherry-picked a bunch of commits that have already been merged into this PR. Can you rebase over master and simply cherry-pick your commits and force push here.

File: analysis/datetime/javatime/javatime.go
Comment (CascadingRadium) [reply analysis/datetime/javatime/javatime.go]: made it iso style since its somewhat similar to iso 8601 spec. Same format is used in swift, ruby etc

**Comment by abhinavdangeti** [comment]
@CascadingRadium I force-pushed all your commits over rebased master to avoid unnecessary cherry-picks of everything in between. Should you need to push more code here, make sure you start from a clean local branch checking-out the latest of `origin/MB-58033` here.

**Review by abhinavdangeti — APPROVED** [review]

**Pull request merged**
